### ISSUE TITLE
yq sometimes drops the first line if it's a comment

### ISSUE THREAD

**Issue by ttung** [mikefarah/yq#1181]
**Describe the bug**
I expect `echo '# abc\n- def\n# ghi' | yq 'sort'` to produce:
```yaml
# abc
- def
# ghi
```

However, it drops the `# abc` and yields
```yaml
- def
# ghi
```

Oddly enough, `yq .` doesn't drop the comment:

```bash
% echo '# abc\n- def\n# ghi' | yq .     
# abc
- def
# ghi
```

Version of yq: 4.21.1
Operating system: mac
Installed via: macports

**Input Yaml**
```yaml
# abc
- def
# ghi
```

**Command**
echo '# abc\n- def\n# ghi' | yq 'sort'
echo '# abc\n- def\n# ghi' | yq .

**Actual behavior**
```yaml
# abc
- def
# ghi
```

**Expected behavior**
```yaml
- def
# ghi
```

**Additional context**
N/A

**Comment by mikefarah** [comment]
Thanks for reporting, will fix in the next release.

**Comment by mikefarah** [comment]
Fixed in 4.24.5

**Comment by daydy16** [comment]
same error when using `| line`
doesnt count the first line, when it's a comment. 
Is this a bug or intentional?
